app/src/main/java/pl/droidevs/books/library/BookActivity.java

```diff
@@ -79,80 +75,82 @@ protected void onCreate(Bundle savedInstanceState) {
         ButterKnife.bind(this);
 
         if (savedInstanceState != null) {
-            bookId = savedInstanceState.getString(EXTRAS_BOOK_ID);
-            imageTransitionName = savedInstanceState.getString(EXTRAS_IMAGE_TRANSITION_NAME);
-            titleTransitionName = savedInstanceState.getString(EXTRAS_TITLE_TRANSITION_NAME);
-            authorTransitionName = savedInstanceState.getString(EXTRAS_AUTHOR_TRANSITION_NAME);
-            shadowTransitionName = savedInstanceState.getString(EXTRAS_SHADOW_TRANSITION_NAME);
-            lastSelectedIndex = savedInstanceState.getInt(EXTRAS_LAST_SELECTED_INDEX);
+            this.animationBundle = savedInstanceState.getBundle(BUNDLE_EXTRAS);
         } else {
-            final Bundle extras = getIntent().getBundleExtra(BUNDLE_EXTRAS);
-            bookId = extras.getString(EXTRAS_BOOK_ID);
-            imageTransitionName = extras.getString(EXTRAS_IMAGE_TRANSITION_NAME);
-            titleTransitionName = extras.getString(EXTRAS_TITLE_TRANSITION_NAME);
-            authorTransitionName = extras.getString(EXTRAS_AUTHOR_TRANSITION_NAME);
-            shadowTransitionName = extras.getString(EXTRAS_SHADOW_TRANSITION_NAME);
-            lastSelectedIndex = extras.getInt(EXTRAS_LAST_SELECTED_INDEX);
+            this.animationBundle = getIntent().getBundleExtra(BUNDLE_EXTRAS);
         }
+
+        setupAnimations();
         setupViewModel();
+    }
 
-        imageView.setTransitionName(imageTransitionName);
-        //ToDo: title transitionName
-        authorTextView.setTransitionName(authorTransitionName);
-        shadowView.setTransitionName(shadowTransitionName);
+    void setupAnimations() {
+        imageView.setTransitionName(animationBundle.getString(EXTRAS_IMAGE_TRANSITION_NAME));
```

Comment:
setup animacji pobierający dane z animationBundle i od razu wrzucam to do widoków :)